src/main/edit_delete/postProcess.ts

```diff
@@ -0,0 +1,66 @@
+import { Transcription } from 'sharedTypes';
+import Clip from './Clip';
+
+/**
+ * Removes deleted words from edited trancription
+ * @param wordList
+ */
+const removeDeleted = (wordList: Array<any>): void => {
+  let i = 0;
+  while (i < wordList.length) {
+    if (wordList[i].deleted) {
+      wordList.splice(i, 1);
+    } else {
+      i += 1;
+    }
+  }
+};
+
+/**
+ * Processes edited data to be ready for export
+ * @param wordList
+ * @returns
+ */
+const FILENAME = 'placeholder_filename.mp4';
+
+const postProcess = (jsonTranscript: Transcription): Array<Clip> => {
+  const wordList = jsonTranscript.words;
+
+  removeDeleted(wordList);
+  const res: Array<Clip> = [];
+  let currentStartTime = -1;
+  let currentDuration = 0;
+
+  for (let i = 0; i < wordList.length - 1; i += 1) {
+    if (currentStartTime === -1) {
+      currentStartTime = wordList[i].startTime;
+      currentDuration = wordList[i].duration;
+    }
+
+    if (
+      wordList[i].startTime + wordList[i].duration ===
+      wordList[i + 1].startTime
+    ) {
+      currentDuration += wordList[i + 1].duration;
+    } else {
+      res.push(new Clip(currentStartTime, currentDuration, FILENAME));
+      currentStartTime = -1;
+      currentDuration = 0;
+    }
+  }
+
+  if (res.length === 0) {
```

Comment:
That would also reduce the need for these cleanups which have a bit of code repetition